websites/S/Sorryfield/presence.ts

```diff
@@ -0,0 +1,156 @@
+const presence = new Presence({
+	clientId: "1016312551958642698",
+});
+
+const enum Assets {
+	LOGO = "large-image",
+	CHART_SEARCH = "chart-search",
+	CHART_HYPORANKED = "chart-hyporanked",
+	CHART_EXPONENTIAL = "chart-exponential",
+	CHART_HARD = "chart-hard",
+	CHART_INTERMEDIATE = "chart-intermediate",
+	CHART_EASY = "chart-easy",
+	CHART_HISTORY = "chart-history",
+	CHART_POSSESION = "chart-possesion",
+	CHART_HOT = "chart-hot",
+	CHART_COLD = "chart-cold",
+	CHART_MULTIPLAYER = "chart-multiplayer",
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		details: "Sorryfield",
+		smallImageKey: Assets.LOGO,
+	};
+	if (
+		document.location.pathname === "/" ||
+		document.location.pathname === "/sorrygle" ||
+		document.location.pathname.startsWith("/song/")
+	) {
+		presenceData.largeImageKey = Assets.LOGO;
+		presenceData.smallImageKey = null;
+	}
+	if (document.location.pathname === "/")
+		presenceData.details = "Viewing Homepage";
+	if (document.location.pathname === "/sorrygle")
+		presenceData.details = "Sorrygle";
+	if (document.location.pathname.startsWith("/song/")) {
+		presenceData.details = "Listening the song";
+		const songName = document
+			.querySelector(".title")
+			.innerHTML.replace('<i class="icon fa-fw fas fa-music"></i>&nbsp;', "")
+			.replace('<i class="icon fa-fw fas fa-lock"></i>&nbsp;', "")
+			.replace('<span class="desc">&nbsp;', " ")
+			.replace("</span>", "")
+			.replace("&amp;", "&");
+		let songArtist = document
+			.querySelector(".song")
+			.children[2].innerHTML.split("&nbsp;")[0];
+		if (document.querySelector(".title").innerHTML.includes("fa-lock"))
+			songArtist = `🔒 ${songArtist}`; // locked music
+		presenceData.state = `${songArtist} - ${songName}`;
+		presenceData.buttons = [
+			{ label: "Listen the song", url: document.location.href },
+		];
+	}
+	if (document.location.pathname === "/java") {
+		if (document.location.href.includes("?")) {
+			presenceData.details = "Java! Singleplay";
+			const chartHeader = document.querySelector(".chart-header");
+			presenceData.state = `${
+				chartHeader.children[2].children[0].innerHTML
+			} Lv.${document.querySelector(".level").innerHTML} / ${
+				chartHeader.children[1].children[1].innerHTML
+			} - ${chartHeader.children[1].children[0].innerHTML}`;
+			presenceData.buttons = [
+				{ label: "Play the chart", url: document.location.href },
+			];
+		} else {
+			presenceData.details = "Java!";
+			presenceData.state = "Selecting chart";
+			const image = document
+				.querySelector(".active")
+				.children[0].innerHTML.replace(
+					'<img class="disc" src="/media/images/chart-',
+					""
+				)
+				.replace('.jpg">', "")
```

Comment:
It may be better to query the image and do the text transformations on the image's src itself, rather than relying on a specific html text structure.